src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Debug/LaunchSettingsJsonEncoding.cs

```diff
@@ -356,12 +356,22 @@ LaunchProfile ReadProfile(string name)
             ReadObject(property =>
             {
                 builder ??= ImmutableArray.CreateBuilder<(string Name, string Value)>();
-                builder.Add((property, ReadString()));
+                builder.Add((property, ReadEnvironmentVariableString()));
             });
 
             return builder?.ToImmutable() ?? ImmutableArray<(string Name, string Value)>.Empty;
         }
 
+        string ReadEnvironmentVariableString()
```

Comment:
Why this special handling only for environment variables? This bug could happen on any string we read.

Reply:
I'd assume because the bug was specifically filed against environment variables. So, without going through and determining what should/shouldn't be handled this way, the specific case is enough to handle the issue at hand. I'd argue that serialization is really easy to mess up when doing it by-hand like this; so, minimizing impact it a good strategy. It might not be the 'right' strategy but makes sense if you're not highly familiar with serialization utilities.